machines/guides-examples/multi-container-machines.html.markerb

```diff
@@ -0,0 +1,341 @@
+
+#### 1. Create cli-config.json
+
+```json
+{
+  "config": {
+  "containers": [
+    {
+      "name": "nginx",
+      "image": "nginx:latest",
+      "files": [
+        {
+          "guest_path": "/etc/nginx/conf.d/default.conf",
+          "local_path": "nginx.conf"
+        }
+      ],
+      "depends_on": [
+        {
+          "container": "echo",
+          "condition": "healthy"
+        }
+      ]
+    },
+    {
+      "name": "echo",
+      "image": "ealen/echo-server",
+      "health_checks": [
+        {
+          "type": "exec",
+          "command": ["wget", "-q", "--spider", "http://localhost"]
+        }
+      ]
+    }
+  ]
+}
+}
+```
+
+#### 2. Update fly.toml
+
+```
+[experimental]
+machine_config = 'cli-config.json'
+container = 'echo'
+
+[http_service]
+internal_port = 8080
+```
+
+The container value determines which container will be replaced by your app image built by `fly deploy`.
+
+#### 3. Run fly deploy
+
+```bash
+fly deploy
+```
+
+This builds and replaces the `echo` container, runs both containers on the same machine, and configures traffic through nginx.
+
+You can also inline the `machine_config` in your `fly.toml` using triple quotes. The first character inside the string must be `{`:
+
+```
+machine_config = '''{
+  "containers": [
+    …
+  ]
+}'''
+```
+
+## Example Configurations
+
+Here are some practical examples of multi-container setups you can use as a starting point:
+
+### Rate Limiter Sidecar
+
+To mitigate excessive traffic, you can run an nginx container as a sidecar to limit requests per second using the `ngx_http_limit_req_module`. Below is a simplified container configuration that enables rate limiting:
+
+```json
+{
+  "config": {
+    "containers": [
+      {
+        "name": "nginx",
+        "image": "nginx:latest",
+        "files": [
+          {
+            "guest_path": "/etc/nginx/conf.d/default.conf",
+            "raw_value": "bGltaXRfcmVxX3pvbmUgJGJpbmFyeV9yZW1vdGVfYWRkciB6b25lPW15X3pvbmU6MTBtIHJhdGU9MXIvczsKCnNlcnZlciB7CiAgbGlzdGVuIDgwODA7CgogIGxvY2F0aW9uIC8gewogICAgbGltaXRfcmVxIHpvbmU9bXlfem9uZTsKICAgIHByb3h5X3Bhc3MgaHR0cDovL2xvY2FsaG9zdDo4MDsKICB9Cn0="
+          }
+        ]
+      },
+      {
+        "name": "echo",
+        "image": "ealen/echo-server"
+      }
+    ]
+  }
+}
+```
+
+The `raw_value` field must contain base64-encoded content when attaching files to containers. In the example above, the NGINX configuration string:
+
+```
+limit_req_zone $binary_remote_addr zone=my_zone:10m rate=1r/s;
+
+server {
+  listen 8080;
+
+  location / {
+    limit_req zone=my_zone;
+    proxy_pass http://localhost:80;
+  }
+}
+```
+
+has been base64-encoded to ensure proper transmission and handling by the Machines API. When the container starts, Fly will decode this value and write the original configuration to the specified `guest_path`.
```

Comment:
Maybe you can move this /tie it in with the preceding line somehow, it looks a little fragmented to have the line start with "has..."